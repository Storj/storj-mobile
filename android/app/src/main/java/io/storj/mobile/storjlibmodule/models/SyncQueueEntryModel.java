package io.storj.mobile.storjlibmodule.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.storj.mobile.storjlibmodule.dataprovider.contracts.SynchronizationQueueContract;


public class SyncQueueEntryModel {
    @Expose
    @SerializedName(SynchronizationQueueContract._ID)
    private int _id;
    @Expose
    @SerializedName(SynchronizationQueueContract._FILE_NAME)
    private String _fileName;
    @Expose
    @SerializedName(SynchronizationQueueContract._LOCAL_PATH)
    private String _localPath;
    @Expose
    @SerializedName(SynchronizationQueueContract._STATUS)
    private int _status;
    @Expose
    @SerializedName(SynchronizationQueueContract._ERROR_CODE)
    private int _errorCode;
    @Expose
    @SerializedName(SynchronizationQueueContract._SIZE)
    private long _size;
    @Expose
    @SerializedName(SynchronizationQueueContract._COUNT)
    private int _count;
    @Expose
    @SerializedName(SynchronizationQueueContract._CREATION_DATE)
    private String _creationDate;
    @Expose
    @SerializedName(SynchronizationQueueContract._BUCKET_ID)
    private String _bucketId;
    @Expose
    @SerializedName(SynchronizationQueueContract._FILE_HANDLE)
    private long _fileHandle;

    public SyncQueueEntryModel(int id, String fileName, String localPath, int status, int errorCode, long size, int count, String creationDate, String bucketId, long fileHandle) {
        _id = id;
        _fileName = fileName;
        _localPath = localPath;
        _status = status;
        _errorCode = errorCode;
        _size = size;
        _count = count;
        _creationDate = creationDate;
        _bucketId = bucketId;
        _fileHandle = fileHandle;
    }

    public boolean isValid() {
        return _fileName == null || _fileName.isEmpty()
                && _localPath == null || _localPath.isEmpty()
                && _bucketId == null || _bucketId.isEmpty(); // FILE SIZE?
    }

    public int getId() { return _id; }
    public String getFileName() { return _fileName; }
    public String getLocalPath() { return _localPath; }
    public int getStatus() { return _status; }
    public int getErrorCode() { return _errorCode; }
    public long getSize() { return _size; }
    public int getCount() { return _count; }
    public String getCreationDate() { return _creationDate; }
    public String getBucketId() { return _bucketId; }
    public long getFileHandle() { return _fileHandle; }
}
