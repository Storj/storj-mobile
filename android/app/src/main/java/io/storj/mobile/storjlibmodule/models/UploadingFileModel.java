package io.storj.mobile.storjlibmodule.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.storj.mobile.storjlibmodule.dataprovider.dbo.UploadingFileDbo;
import io.storj.mobile.storjlibmodule.dataprovider.contracts.UploadingFileContract;

/**
 * Created by Yaroslav-Note on 3/7/2018.
 */

public class UploadingFileModel {
    @Expose
    @SerializedName(UploadingFileContract._FILE_HANDLE)
    private long _fileHandle;
    @Expose
    @SerializedName(UploadingFileContract._PROGRESS)
    private double _progress;
    @Expose
    @SerializedName(UploadingFileContract._SIZE)
    private long _size;
    @Expose
    @SerializedName(UploadingFileContract._UPLOADED)
    private long _uploaded;

    @Expose
    @SerializedName(UploadingFileContract._NAME)
    private String _name;
    @Expose
    @SerializedName(UploadingFileContract._URI)
    private String _uri;
    @Expose
    @SerializedName(UploadingFileContract._BUCKET_ID)
    private String _bucketId;

    public UploadingFileModel(long fileHandle, long size, String name, String uri, String bucketId) {
        this(fileHandle, 0, size, 0, name, uri, bucketId);
    }

    public UploadingFileModel(UploadingFileDbo uploadingFileDbo) {
        this(uploadingFileDbo.getId(),
                uploadingFileDbo.getProgress(),
                uploadingFileDbo.getSize(),
                uploadingFileDbo.getUploaded(),
                uploadingFileDbo.getName(),
                uploadingFileDbo.getUri(),
                uploadingFileDbo.getBucketId());
    }

    public UploadingFileModel(long fileHandle, double progress, long size, long uploaded, String name, String uri, String bucketId) {
        _fileHandle = fileHandle;
        _progress = progress;
        _size = size;
        _uploaded = uploaded;
        _name = name;
        _uri = uri;
        _bucketId = bucketId;
    }

    public boolean isValid() {
        return  _name != null && !_name.isEmpty() &&
                _uri != null && !_uri.isEmpty() &&
                _bucketId != null && !_bucketId.isEmpty();
    }

    public long getFileHandle() {
        return _fileHandle;
    }
    public double getProgress() {
        return _progress;
    }
    public long getSize() {
        return _size;
    }
    public long getUploaded() {
        return _uploaded;
    }

    public String getName() {
        return _name;
    }
    public String getUri() {
        return _uri;
    }
    public String getBucketId() {
        return _bucketId;
    }
}
