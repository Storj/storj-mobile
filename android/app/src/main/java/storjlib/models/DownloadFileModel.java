package storjlib.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Crawter on 27.02.2018.
 */

public class DownloadFileModel {
    @Expose
    @SerializedName("fileId")
    private String _fileId;
    @Expose
    @SerializedName("localPath")
    private String _localPath;

    public DownloadFileModel(String fileId, String localPath) {
        _fileId = fileId;
        _localPath = localPath;
    }
}
