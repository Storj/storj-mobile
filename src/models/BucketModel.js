import ItemModel from './ItemModel';

export default class BucketModel extends ItemModel {
    /**
    * Describes bucket entity
    * @param {object} bucket
    */
    constructor(bucket) {
        if( bucket
            && bucket.name
            && bucket.id
            && Number.isInteger(bucket.hash)
            && typeof(bucket.isDecrypted) === "boolean"
            && bucket.created) {

            super(bucket.name, bucket.id, bucket.created);
            
            this.hash = bucket.hash;
            this.isDecrypted = bucket.isDecrypted;
            this.created = bucket.created;
        } else {
            super();

            this.hash = 0;
            this.isDecrypted = false;
            this.created = null;
        }
    };
}