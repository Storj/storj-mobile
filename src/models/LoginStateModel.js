export class LoginStateModel {
    /**
    * State model used for storing email and password in LoginContainer. For internal usage, not for reducer or store
    * @param {string} - email
    * @param {string} - password
    * @param {string} - mnemonic
    */
    constructor(email, password, mnemonic, passCode) {
        this.email = email ? email : '';
        this.password = password ? password : '';
        this.mnemonic = mnemonic ? mnemonic : '';
        this.passCode = passCode ? passCode : '';
    };

    /**
     * Helper function for comparing two instances. Returns false
     *  if object properties differs from those described in this model.
     * @param {LoginStateModel} stateModel 
     * @returns {Boolean} 
     */
    isEqualTo(stateModel) {
        let props = Object.getOwnPropertyNames(stateModel);

        if(!(props.length === 4 && props.includes('email', 'password', 'mnemonic', 'passcode')))
            return false;          
            
        return(
            stateModel.email === this.email 
            && stateModel.password === this.password
            && stateModel.mnemonic === this.mnemonic
            && stateModel.passCode === this.passCode
        );
    };
}