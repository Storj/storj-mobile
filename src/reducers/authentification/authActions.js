import { NavigationActions } from 'react-navigation';
import authActions from '../../utils/constants/actionConstants';
import { changePasswordRequest } from '../../utils/dataservice';

const { LOGIN, LOGIN_SUCCESS, LOGIN_ERROR, 
    REGISTER, REGISTER_SUCCESS, REGISTER_ERROR, 
    SET_EMAIL_NOT_CONFIRMED, SET_EMAIL_CONFIRMED,
    SET_ACCOUNT_NOT_EXIST, SET_ACCOUNT_EXIST } = authActions;


/**
 * sends email with link to reset password
 * @param {string} email registered email
 * @returns validity of request
 */
export async function resetPassword(email) {
    let request = changePasswordRequest(email);

    let response = await fetch(request);
    
    return response.ok;
};

/**
 * ActionCreator on login success 
 * @returns action
 */
function loginSuccess() {
    return { type: LOGIN_SUCCESS };
};

/**
 * ActionCreator on login error
 * @returns action 
 */
function loginError() {
    return { type: LOGIN_ERROR };
};

/**
 * ActionCreator on login that changing authreducer userInfo state
 * @param {string} email registered email
 * @param {string} password registered password 
 * @param {string} mnemonic registered mnemonic
 * @returns action
 */
function login(email, password, mnemonic) {
    return { type: LOGIN, payload: { email, password, mnemonic } };
};

/**
 * ActionCreator that provides redux navigation to RegisterScreen
 * @returns action
 */
function navigateToRegisterScreen() {
    return NavigationActions.navigate({ routeName: 'RegisterScreen'});
};

/**
 * ActionCreator on register that changing authreducer userInfo state
 * @param {string} email 
 * @param {string} password 
 * @param {string} mnemonic 
 * @returns action
 */
function register(email, password) {
    return { type: REGISTER, payload: { email, password } };
};

/**
 * ActionCreator on register success 
 * @returns action
 */
function registerSuccess(mnemonic) {
    return { type: REGISTER_SUCCESS, payload: { mnemonic }};
};

/**
 * ActionCreator on register error that setting error property in authREducer state in errorMessage
 * @param {string} error error massage 
 * @returns action
 */
function registerError(error) {
    return { type: REGISTER_ERROR, error };
};

/**
 * ActionCreator that provides redux back navigation
 * @returns action
 */
function navigateBack() {
    return NavigationActions.back();
};

/**
 * ActionCreator that provides redux navigation to LoginScreen
 * @returns action
 */
function redirectToLoginScreen() {
    return NavigationActions.reset({
        index: 0,
        actions: [
          NavigationActions.navigate({ routeName: 'LoginScreen'})
        ]
    });
};

/**
 * ActionCreator that provides redux navigation to MainScreen
 * @returns action
 */
function redirectToMainScreen() {
    return NavigationActions.reset({
        index: 0,
        actions: [
          NavigationActions.navigate({ routeName: 'MainScreen'})
        ]
    });
};

/**
 * ActionCreator that provides redux navigation to QRScannerScreen
 * @returns action
 */
function redirectToQRScannerScreen() {
    return NavigationActions.navigate({ routeName: 'QRScannerScreen' })
};

/**
 * ActionCreator that provides redux navigation to TermsOfUseScreen
 * @returns action
 */
function redirectToTermsOfUse() {
    return NavigationActions.navigate({ routeName: 'TermsOfUseScreen' })
};

/**
 * ActionCreator that provides redux navigation to AuthFailureInfoScreen
 * @param {string} error
 * @returns action
 */
function redirectToAuthFailureScreen(params) {
    return NavigationActions.navigate({ routeName: 'AuthFailureInfoScreen', params });
};

/**
 * ActionCreator that provides redux navigation to RegisterSuccessInfoScreen
 * @returns action
 */
function redirectToRegisterSuccessScreen() {
    return NavigationActions.navigate({ routeName: 'RegisterSuccessInfoScreen' });
};

/**
 * ActionCreator that provides redux navigation to InitializationScreen
 * @returns action
 */
function redirectToInitializeScreen() {
    return NavigationActions.navigate({ routeName: 'InitializationScreen' });
}

function setEmailNotConfirmed() {
    return { type: SET_EMAIL_NOT_CONFIRMED }
}

function setEmailConfirmed() {
    return { type: SET_EMAIL_CONFIRMED }
}

function setAccountNotExist() {
    return { type: SET_ACCOUNT_NOT_EXIST }
}

function setAccountExist() {
    return { type: SET_ACCOUNT_EXIST }
}

// ActionCreators for LoginContainer
export const loginActionsCreators = {
    setAccountNotExist,
    setAccountExist,
    setEmailNotConfirmed,
    setEmailConfirmed,
    loginSuccess,
    loginError,
    login,
    navigateToRegisterScreen,
    redirectToAuthFailureScreen,
    redirectToMainScreen,
    redirectToInitializeScreen,
    redirectToQRScannerScreen
};

// ActionCreators for RegisterContainer
export const registerActionsCreators = {
    registerSuccess,
    registerError,
    register,
    navigateBack,
    redirectToTermsOfUse,
    redirectToLoginScreen,
    redirectToAuthFailureScreen,
    redirectToRegisterSuccessScreen
}
export const qrScannerActionCreators = {
    loginSuccess,
    loginError,
    login,
    navigateToRegisterScreen,
    redirectToAuthFailureScreen,
    redirectToMainScreen,
    navigateBack,
    redirectToInitializeScreen
}
