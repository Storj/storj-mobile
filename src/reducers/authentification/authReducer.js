import loginActions from '../../utils/constants/actionConstants';

const { LOGIN, LOGIN_SUCCESS, LOGIN_ERROR, REGISTER, REGISTER_SUCCESS, REGISTER_ERROR } = loginActions;

const initialState = { user: { isLoggedIn: false, email: null, password: null, mnemonic: null,
                                passCode: null, error: null, isLoading: false, isRedirectedFromRegister: false } };


export default function authReducer(state, action) {

    let newState = { user: {} };
    if(state && state.user) {
        newState.user = Object.assign({}, state.user);
    }

    //TODO: we can call return after switch, and reduce lines of code
    switch(action.type) {
        case LOGIN_SUCCESS:
            newState.user.isLoggedIn = true;

            return newState;

        case LOGIN_ERROR: 
            //TODO: should be removed. 
            //Logout action will set isLoggedIn ==> false. 
            //So here we just setting false into false. Redundant
            newState.user.isLoggedIn = false;

            return newState;

        case LOGIN:
            newState.user.email = action.payload.email;
            newState.user.password = action.payload.password;
            newState.user.mnemonic = action.payload.mnemonic;
            newState.user.passCode = action.payload.passCode;

            return newState;
        
        case REGISTER:
            newState.user.isLoading = true;
            newState.user.email = action.payload.email;
            newState.user.password = action.payload.password;
            
            return newState;

        case REGISTER_SUCCESS:
            newState.user.mnemonic = action.payload.mnemonic;
            newState.user.isLoading = false;
            newState.user.isRedirectedFromRegister = true;

            return newState;

        case REGISTER_ERROR:
            newState.user.error = action.error;
            newState.user.email = null;
            newState.user.password = null;
            newState.user.mnemonic = null;
            newState.user.IsRedirectedFromRegister = false;

            return newState;
            
        default:
            return state || initialState;
    }
};