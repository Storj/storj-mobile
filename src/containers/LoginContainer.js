import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { loginActionsCreators } from '../reducers/authentification/authActions';
import StorjLib from '../utils/StorjModule';
import LoginComponent from '../components/LoginComponent';
import validator from '../utils/validator';
import { LoginStateModel } from '../models/LoginStateModel';
import { LoginErrorModel } from '../models/LoginErrorModel';
import infoScreensConstants from '../utils/constants/infoScreensConstants';
import { authConstants } from '../utils/constants/storageConstants';
import { getEmail, getMnemonic, getPassword, getFirstAction, setFirstAction } from '../utils/AsyncStorageModule';

/**
 * Container for LoginComponent
 */
class LoginContainer extends Component {
	constructor(props) {
        super(props);
        
        this.state = {
            stateModel: new LoginStateModel(),
            errorModel: new LoginErrorModel(),
            isLoading: false
        };
    };

    /**
     * Hiding navigation header
     */
    static navigationOptions = {
        header: null
    };

    /**
     * Fill local state from store on component did mount
     */
    async componentDidMount() {
        let email, password, mnemonic;

        if(this.props.user.mnemonic) {
            email = this.props.user.email;
            password = this.props.user.password;
            mnemonic = this.props.user.mnemonic;
        } else {
            email = await getEmail(),
            password = await getPassword(),
            mnemonic = await getMnemonic()   
        }

        this.setState({
            stateModel: new LoginStateModel(email, password, mnemonic)
        });
    };

    /**
     * Changing internal state when user login inputting
     * @param {string} value current value in Input
     */
    onChangeEmailInput(value) {
        this.setState({
            stateModel: new LoginStateModel(
                value,
                this.state.stateModel.password,
                this.state.stateModel.mnemonic
            )
        });
    };

    /**
     * Changing internal state when user password inputting
     * @param {string} value current value in Input
     */
    onChangePasswordInput(value) {
        this.setState({
            stateModel: new LoginStateModel(
                this.state.stateModel.email,
                value,
                this.state.stateModel.mnemonic
            )
        });
    };
    /**
     * Changing internal state when user password inputting
     * @param {string} value current value in Input
     */
    onChangeMnemonicInput(value) {
        this.setState({
            stateModel: new LoginStateModel(
                this.state.stateModel.email,
                this.state.stateModel.password,
                value
            )
        });
    };

    /**
     * Handle if was allready in use
     */
    async handleFirstLaunch() {
        if(!await getFirstAction()) {
            await setFirstAction();
        }
    };

    /**
     * try validate login as email and invokes actionCreators  
     * to change userInfo in store
     */
	async tryLogin() {
        if(this.state.isLoading) return;
        this.setState({ isLoading: true });

        let isEmailValid = validator.isEmail(this.state.stateModel.email);
        let isPasswordValid = this.state.stateModel.password ? true : false;
        let isMnemonicValid = await StorjLib.checkMnemonic(this.state.stateModel.mnemonic);

        if(isEmailValid && isPasswordValid && isMnemonicValid) {
            await this.login();
        } else {
            this.setState({
                errorModel: new LoginErrorModel(
                    !isEmailValid,
                    !isPasswordValid,
                    !isMnemonicValid,
                    this.state.errorModel.isCredentialsError
                )
            });
        }

        this.setState({ isLoading: false });
    };


    async login() {
        this.props.login(this.state.stateModel.email, 
                         this.state.stateModel.password,
                         this.state.stateModel.mnemonic);

        let areCredentialsValid = await StorjLib.verifyKeys(
            this.state.stateModel.email, 
            this.state.stateModel.password);

        if(!areCredentialsValid) {
            this.setState({
                errorModel: new LoginErrorModel(
                    this.state.errorModel.isEmailError,
                    this.state.errorModel.isPasswordError,
                    this.state.errorModel.isMnemonicError,
                    !areCredentialsValid
                )
            });

            this.props.loginError();
            this.props.redirectToAuthFailureScreen({
                 mainText: infoScreensConstants.loginFailureMainText, 
                 additionalText: infoScreensConstants.loginFailureAdditionalText 
            });

            return;
        }

        let areKeysImported = await StorjLib.importKeys(
            this.state.stateModel.email,
            this.state.stateModel.password,
            this.state.stateModel.mnemonic
        );
        
        if(areKeysImported) {
            await this.handleFirstLaunch();
            this.props.loginSuccess();
            this.props.redirectToInitializeScreen();
        } else {
            this.props.loginError();
            this.props.redirectToAuthFailureScreen({ 
                mainText: infoScreensConstants.loginFailureMainText, 
                additionalText: infoScreensConstants.loginFailureAdditionalText 
            });
        }
    };

    /**
     * invokes actionCreators that provides navigations
     */
    redirectToRegisterScreen() {
		this.props.navigateToRegisterScreen();
    };
    redirectToMainPageScreen() {
		this.props.redirectToMainScreen();
    };

	render() {
		return(
            <LoginComponent
                isLoading = { this.state.isLoading }
                email = { this.state.stateModel.email }
                password = { this.state.stateModel.password }
                mnemonic = { this.state.stateModel.mnemonic }
                isRedirectedFromRegister = { this.props.user.isRedirectedFromRegister }
                isEmailError = { this.state.errorModel.isEmailError }
                isPasswordError = { this.state.errorModel.isPasswordError }
                isMnemonicError = { this.state.errorModel.isMnemonicError }
                areCredentialsValid = { this.state.errorModel.areCredentialsValid }
                onChangeLogin = { this.onChangeEmailInput.bind(this) }
                onChangePassword = { this.onChangePasswordInput.bind(this) }
                onChangeMnemonic = { this.onChangeMnemonicInput.bind(this) }
                onSubmit = { this.tryLogin.bind(this) }
                registerButtonOnPress = { this.redirectToRegisterScreen.bind(this) } />
		);
	};
}

/**
 * connecting reducer to component props 
 */
function mapStateToProps(state) { return { user: state.authReducer.user }; };
function mapDispatchToProps(dispatch) { return bindActionCreators(loginActionsCreators, dispatch); };

/**
 * Creating LoginScreen container
 */
export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);

/**
 * Checking RegisterContainer correct prop types
 */
LoginContainer.propTypes = {
    user: PropTypes.shape({
        isLoggedIn: PropTypes.bool,
        email: PropTypes.string,
        password: PropTypes.string,
        mnemonic: PropTypes.string,
        isLoading: PropTypes.bool,
        error: PropTypes.string
    })
};
