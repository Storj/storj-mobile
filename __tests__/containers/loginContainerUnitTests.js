import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import LoginContainer from '../../src/containers/LoginContainer';
import configureStore from 'redux-mock-store';
import { loginActionsCreators, registerActionsCreators } from '../../src/reducers/authentification/authActions';
import authReducer from '../../src/reducers/authentification/authReducer';
import { testUser } from 'jest-mock';
import sinon from 'sinon';
import { LoginStateModel } from '../../src/models/LoginStateModel';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const initialState = { authReducer : { user: { isLoggedIn: false, email: null, password: null }} };

describe('LoginContainer', ()=> {

	it('renders correctly', () => {

		const mockedStore = mockStore(initialState);
		const wrapper = shallow(
			<LoginContainer	
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
			{ context: { store: mockedStore } }
		);

		expect(wrapper.dive()).toMatchSnapshot();
	});

	it('triggers dispatch and login once on tryLogin method call', () => {
		const mockedStore = mockStore(initialState);
		const storeSpy = sinon.spy(mockedStore, 'dispatch');

		const wrapper = shallow(
			<LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
			{ context: { store: mockedStore } }
		);

		const handle = wrapper.dive().instance();
		const loginSpy = sinon.spy(handle, 'login');

		handle.login();

		expect(storeSpy.called).toBe(true);
		expect(loginSpy.calledOnce).toBe(true);
	});    

	it('triggers setState on onChangeEmailInput', () => {
		const mockedStore = mockStore(initialState);
		const wrapper = shallow(
			<LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
			{ context: { store: mockedStore } }
		);

		const handle = wrapper.dive().instance();
		const stateSpy = sinon.spy(handle, 'setState');

		handle.onChangeEmailInput(handle);

		expect(stateSpy.calledOnce).toBe(true);
	});    

	it('triggers setState on onChangePasswordInput', () => {
		const mockedStore = mockStore(initialState);
		const wrapper = shallow(
			<LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
			{ context: { store: mockedStore } }
		);

		const handle = wrapper.dive().instance();
		const stateSpy = sinon.spy(handle, 'setState');

		handle.onChangePasswordInput();

		expect(stateSpy.calledOnce).toBe(true);
	});   

	it('triggers navigateToRegisterScreen on redirectToRegisterScreen', () => {
		const mockedStore = mockStore(initialState);
		const wrapper = shallow(
			<LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
			{ context: { store: mockedStore } }
		);

		const handle = wrapper.dive().instance();
		const redirectSpy = sinon.spy(handle, 'redirectToRegisterScreen');

		handle.redirectToRegisterScreen();

		expect(redirectSpy.calledOnce).toBe(true);
	}); 

	it('triggers setState on ChangePassword', () => {
		const mockedStore = mockStore(initialState);
        const wrapper = shallow(
			<LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
            { context: { store: mockedStore } }
        );
        const newPass = 'newPassword';
    
        const handle = wrapper.dive().instance();
        const prevState = handle.state;
        const newState = new LoginStateModel(prevState.email, newPass);

        handle.onChangePasswordInput(newPass);

        expect(newState.isEqualTo(handle.state.stateModel)).toBe(true);
        expect(handle.state.stateModel.password).toEqual(newPass);
    });  

    it('triggers setState on ChangeEmail', () => {
		const mockedStore = mockStore(initialState);
		
        const wrapper = shallow(
            <LoginContainer 
				screenProps = {{ redirectToMnemonicHelpScreen: () => {}}} />,
            { context: { store: mockedStore } }
        );

        const newEmail = 'new@gmail.com';
    
        const handle = wrapper.dive().instance();
        const prevState = handle.state;
        const newState = new LoginStateModel(newEmail, prevState.password);

        handle.onChangeEmailInput(newEmail);

        expect(newState.isEqualTo(handle.state.stateModel)).toBe(true);
        expect(handle.state.stateModel.email).toEqual(newEmail);
    });
});

