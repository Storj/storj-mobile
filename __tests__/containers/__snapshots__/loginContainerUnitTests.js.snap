// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginContainer
    login={[Function]}
    loginError={[Function]}
    loginSuccess={[Function]}
    navigateToRegisterScreen={[Function]}
    redirectToAuthFailureScreen={[Function]}
    redirectToInitializeScreen={[Function]}
    redirectToMainScreen={[Function]}
    redirectToQRScannerScreen={[Function]}
    setAccountExist={[Function]}
    setAccountNotExist={[Function]}
    setEmailConfirmed={[Function]}
    setEmailNotConfirmed={[Function]}
    user={
      Object {
        "email": null,
        "isLoggedIn": false,
        "password": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "areCredentialsValid": undefined,
      "email": "",
      "isEmailError": false,
      "isLoading": false,
      "isMnemonicError": false,
      "isPasswordError": false,
      "isRedirectedFromRegister": undefined,
      "mnemonic": "",
      "onChangeLogin": [Function],
      "onChangeMnemonic": [Function],
      "onChangePassword": [Function],
      "onSubmit": [Function],
      "password": "",
      "redirectToQRScannerScreen": [Function],
      "registerButtonOnPress": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "areCredentialsValid": undefined,
        "email": "",
        "isEmailError": false,
        "isLoading": false,
        "isMnemonicError": false,
        "isPasswordError": false,
        "isRedirectedFromRegister": undefined,
        "mnemonic": "",
        "onChangeLogin": [Function],
        "onChangeMnemonic": [Function],
        "onChangePassword": [Function],
        "onSubmit": [Function],
        "password": "",
        "redirectToQRScannerScreen": [Function],
        "registerButtonOnPress": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
